File: hw5/to_csv.py
# Kay Sweebe
# CS471 - S. Cooper
# Programming Assignment 5
#
# File conversion to CSV


# Placeholder to store what will be written to a file later.
# Need to look at instructions
csv_text = ""

# Open file to read
filepath = 'test.txt'
try:
    with open(filepath, 'r') as fp:

        # Read file
        line = fp.readline()

        while line:

            # Start of File and End of File lines
            if line != '%%%%%\n' and line != '$$$$$\n':

                # Declare a reformatted line for the line in the file
                reformatted_line = line[:-1]

                # Look for commas
                if line.find(",") != -1: #found

                    reformatted_line = "\"" + line[:-1] + "\"" # -1 to remove newline

                # Look for all double quotes
                if line.find("\"") != -1: #found first one
                    reformatted_line = list(reformatted_line)                
                    for i, c in enumerate(reformatted_line):
                        if c == "\"":
                            reformatted_line[i] = "\"\""
                
                # Add line to output
                csv_text = csv_text + ''.join(reformatted_line)
                csv_text = csv_text + ", "

            # Get next line for loop
            line = fp.readline()

except Exception as e:
    print(e)

# Write CSV
try:
    with open("csv_python.csv", 'w+') as fq:
        fq.write(csv_text[:-2])
except Exception as e:
    print(e)

File: hw5/toCSV.java
/*                                                                          
 * Kay Sweebe
 * CS471 - S. Cooper
 * Ch. 5 Programming Assignment
 *
 * Purpose:
*/
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.util.Scanner;

class toCSV{
   public static void main(String args[])
     throws FileNotFoundException
   {
     // Make file object
     File file = new File("test.txt");

     // Make scanner object
     Scanner sc = new Scanner(file);

     // Make string to write to file
     String csv_string = "";

     // Read file by line
     while (sc.hasNextLine()){

       // Grab line
       String line = sc.nextLine();

       // Edited line
       String edit_line = line;

       // Edit all lines that are not the first or last
       if( !line.equals("%%%%%") && !line.equals("$$$$$") ){

         // Look for double quotes
         if(line.contains("\"")){
           edit_line = line.replaceAll("\"", "\"\"");
         }// end if quotes

         // Look for commas
         if(line.contains(",")){
           edit_line = "\"" + edit_line + "\"";
         }// end if comma

         csv_string = csv_string + edit_line + ", ";
       }// end if beginning or ending line
     }// end while

     // Remove last comma
     csv_string = csv_string.substring(0, csv_string.length()-2);

     System.out.println(csv_string);

     // Write csv
     try{
       FileWriter fw = new FileWriter("csv_java.csv");
       fw.write(csv_string);
       fw.close();
     }catch(Exception e){
       System.out.println(e);
     }// end write csv
   }// end main
 }// end class
